tests/providers/aws/aws_e2e.py

```diff
@@ -56,99 +52,139 @@ def setUpClass(cls):
       raise unittest.SkipTest(str(exception))
     cls.instance_to_analyse = project_info['instance']
     cls.zone = project_info['zone']
-    cls.volume_to_forensic = project_info.get('volume_id', None)
+    cls.dst_zone = project_info.get('destination_zone', None)
+    cls.volume_to_copy = project_info.get('volume_id', None)
+    cls.encrypted_volume_to_copy = project_info.get('encrypted_volume_id', None)
     cls.aws = account.AWSAccount(cls.zone)
     cls.analysis_vm_name = 'new-vm-for-analysis'
     cls.analysis_vm, _ = forensics.StartAnalysisVm(cls.analysis_vm_name,
                                                    cls.zone,
                                                    10)
-    cls.volumes = []
+    cls.volumes = []  # List of (AWSAccount, AWSVolume) tuples
 
-  def test_end_to_end_boot_volume(self):
+  def testBootVolumeCopy(self):
     """End to end test on AWS.
 
-    This end-to-end test runs directly on AWS and tests that:
-      1. The account.py module connects to the target instance and makes a
-          snapshot of the boot volume (by default) or of the volume passed in
-          parameter to the forensics.CreateVolumeCopy() method.
-      2. A new volume is created from the taken snapshot.
+    Test copying the boot volume of an instance.
     """
 
-    # Make a copy of the boot volume of the instance to analyse
-    boot_volume_copy = forensics.CreateVolumeCopy(
+    volume_copy = forensics.CreateVolumeCopy(
         self.zone,
         instance_id=self.instance_to_analyse
         # volume_id=None by default, boot volume of instance will be copied
     )
+    self._StoreVolumeForCleanup(self.aws, volume_copy.volume_id)
+    self.assertEqual(self.volumes[-1][1].volume_id, volume_copy.volume_id)
 
-    # The volume copy should be attached to the AWS account
-    self.volumes.append(
-        self.aws.ResourceApi(EC2_SERVICE).Volume(boot_volume_copy.volume_id))
-    self.assertEqual(self.volumes[-1].volume_id, boot_volume_copy.volume_id)
+  def testVolumeCopy(self):
+    """End to end test on AWS.
+
+    Test copying a specific volume.
+    """
+
+    if not self.volume_to_copy:
+      return
 
-  def test_end_to_end_other_volume(self):
+    volume_copy = forensics.CreateVolumeCopy(
+        self.zone, volume_id=self.volume_to_copy)
+    self._StoreVolumeForCleanup(self.aws, volume_copy.volume_id)
+    self.assertEqual(self.volumes[-1][1].volume_id, volume_copy.volume_id)
```

Comment:
I'm not sure I understand what's going on here. You're comparing the volume_id attribute of the volume you just appended to itself? `CreateVolumeCopy` could just return a random volume object and the test would still pass.

I think a better e2e test would be to call `CreateVolumeCopy`, then try to confirm that this had the action you wanted in the remote AWS project, either:

* through a unique attribute set server-side by AWS in the response
* or by listing the remove volumes and confirming that the volume ID that you initially received is there

Reply:
Ah yeah, that doesn't make much sense. I changed that!